include/power/power.h

```diff
@@ -314,6 +315,44 @@ void pm_notify_power_state_entry(enum power_states state);
  */
 void pm_notify_power_state_exit(enum power_states state);
 
+/**
+ * @brief Set constraint for a power states.
+ *
+ * A constraint will prevents the system to go to a specific power state.
```

Comment:
Document that setting and releasing constraints must be done from a thread context.